### PR TITLE
Clean up some unnecessary closures in pwndbg.commands

### PR DESCRIPTION
**Pull request by zachriggle**
There are a bunch of closures used for decorating, because I didn't have a good grasp on classes-as-decorators when I originally wrote this code.

### CODE REVIEW

**Comment by disconnect3d** [comment]
Having a stupid simple command:
```
parser = argparse.ArgumentParser(description='lala')
parser.add_argument('bar', type=int)
@ArgparsedCommand(parser)
def foo(bar):
    print(bar)
```

Could have a better output:
```
pwndbg> foo
usage: foo [-h] bar
foo: error: the following arguments are required: bar
Python Exception <class 'SystemExit'> 2: 
Error occurred in Python command: 2
```

Would be nice to have it just saying `foo: the following arguments are required: bar`.

**Comment by zachriggle** [comment]
I'm not sure why the last two lines are showing up.  Where are they originating from?

**Comment by disconnect3d** [comment]
Fun-sad-fact: both `set exception-debugger on` and 'set exception-verbose on` and `foo` afterwards exits gdb :(

**Comment by zachriggle** [comment]
Try now ^_^

**Comment by disconnect3d** [comment]
Hm...

There are some weird/bug prone situations:
```
pwndbg> i r rsp
rsp            0x7fffffffd850	0x7fffffffd850
pwndbg> foo $rsp
140737488345168
pwndbg> i r esp
esp            0xffffd850	-10160
pwndbg> foo $esp
-10160
```

The `$esp` here passed as an address to command, casted/interpreted by us as int instead of unsigned, probably might boom in some cases.

On the other hand, one can do:
```
pwndbg> foo "(unsigned) $esp"
4294957136
```


When we try to get a register that doesn't exist, there is not really nice output:
```
pwndbg> foo $bar
Python Exception <class 'gdb.error'> Invalid cast.: 
Error occurred in Python command: Invalid cast.
```

Also, as those are gdb expressions, arithmetic works, but may not be obvious all the time (quotation required):
```
pwndbg> foo 1+1
2
pwndbg> foo 1 + 1
usage: foo [-h] bar
foo: error: unrecognized arguments: + 1
```

File: pwndbg/commands/__init__.py [hunk 1]
```diff
@@ -37,16 +37,28 @@ def __init__(self, function, inc=True, prefix=False):
         self.__doc__ = function.__doc__
 
     def split_args(self, argument):
-        return gdb.string_to_argv(argument)
+        """Split a command-line string from the user into arguments.
+
+        Returns:
+            A tuple / dict, in the form of ``*args, **args``.
```
Comment (disconnect3d) [inline pwndbg/commands/__init__.py]: copy paste bug, should be: `*args, **kwargs`

**Comment by zachriggle** [comment]
Regarding your arithmetic example, there is **not any way at all** to handle the string "1 + 1" (with spaces) without some custom parsing magic.  `gdb.string_to_argv` splits this into three distinct tokens.  Users will need to remove any spaces.

**Comment by zachriggle** [comment]
I added a bit to enhance the error messages slightly.

```
pwndbg> checksec
checksec: There is no file loaded.
pwndbg> hexdump $eax
hexdump: No registers. (<class 'gdb.error'>)
pwndbg> hexdump $rax
hexdump: Invalid cast. (<class 'gdb.error'>)
pwndbg> hexdump 0xcafe
+0000 0x00cafe
pwndbg> hexdump *(int*)0xcafe
hexdump: Cannot access memory at address 0xcafe (<class 'gdb.MemoryError'>)
```

**Review by disconnect3d — APPROVED** [review]

**Pull request merged**